src/core/defs.h

```diff
@@ -85,30 +85,30 @@ typedef struct {
 #define NNI_ALLOC_STRUCTS(s, n) nni_zalloc(sizeof(*s) * n)
 #define NNI_FREE_STRUCTS(s, n) nni_free(s, sizeof(*s) * n)
 
-#define NNI_PUT16(ptr, u)                                    \
```

Comment:
I don't really understand the motivation for this change.  The use of "u" as a suffix is unambiguous with this (it always is prefixed by numeric digits!) -- if this was causing some compiler grief I'd like to see compiler output, because it strongly indicates a compiler defect (and a serious one at that.)

Reply:
Hi, yes, this is not seen in any compiler newer than MSVC 1800.

```
cmake -G "Visual Studio 12 2013" -A Win32 ..
```
Error (one example):
```
  tls.c
C:\github\nng\src\transport\tls\tls.c(584): error C2059: syntax error : 'bad suffix on number' [C:\github\nng\build\src
\nng.vcxproj]
C:\github\nng\src\transport\tls\tls.c(584): error C2146: syntax error : missing ')' before identifier 'p' [C:\github\nn
g\build\src\nng.vcxproj]
C:\github\nng\src\transport\tls\tls.c(584): error C2059: syntax error : ')' [C:\github\nng\build\src\nng.vcxproj]
C:\github\nng\src\transport\tls\tls.c(585): warning C4293: '>>' : shift count negative or too big, undefined behavior [
C:\github\nng\build\src\nng.vcxproj]
```

Fixed by not using `u` as macro parameter.

Reply:
Wow.  That's shocking.  Completely and totally broken behavior.  I guess it also means you can't have a variable named "u".

Please submit this as a separate PR.

Reply:
Looking in to it a bit more, suffixing with upper case `U` works together with a lowercase macro variable named lower case `u`.

As `Letters in the integer literals are case-insensitive`, I'll make a separate PR with `u`'s and `U`'s.